### PR TITLE
[envoy] Add support for dictionaries 

### PR DESCRIPTION
**Pull request by asraa**
Moves dictionary with form <fuzz_target>.dict to /out/, oss-fuzz automatically picks up dictionaries of this form.

@htuch 

Signed-off-by: Asra Ali <asraa@google.com>

### CODE REVIEW

**Comment by asraa** [comment]
Update: we decided on envoy upstream we may want to have multiple dictionary files, so before merging I'll update this to concatenate a list of *.dict files in to the <target>.dict file and move that . Marking PR as WIP.

**Comment by Dor1s** [comment]
Ack!

**Comment by asraa** [comment]
Hey! OSS-Fuzz handles multiple dictionaries now (tested locally as well), so this is ready for review now.

**Comment by Dor1s** [comment]
@asraa do we merge it, or you need anyone from Envoy to take a look as well?

File: projects/envoy/build.sh [hunk 1]
```diff
@@ -119,6 +119,11 @@ do
   mkdir -p "${CORPUS_UNTAR_PATH}"
   tar -C "${CORPUS_UNTAR_PATH}" -xvf bazel-bin/"${t}"_corpus_tar.tar
   TARGET_BASE="$(expr "$t" : '.*/\(.*\)_fuzz_test')"
+  # There may be *.dict files in this folder that need to be moved into the OUT dir.
+  for dictionary in "${CORPUS_UNTAR_PATH}"/*.dict; do
+    [ -f "$dictionary" ] || continue
```
Comment (htuch) [inline projects/envoy/build.sh]: Why is this needed? When would we hit something matching `*.dict` that isn't a regular file?

Comment (asraa) [reply projects/envoy/build.sh]: Hm, just to make sure there are dictionaries. if there's not `*.dict` then mv will log a cannot stat. I think I'll just make it a one-liner find with -exec mv.

File: projects/envoy/build.sh [hunk 2]
```diff
@@ -119,6 +119,8 @@ do
   mkdir -p "${CORPUS_UNTAR_PATH}"
   tar -C "${CORPUS_UNTAR_PATH}" -xvf bazel-bin/"${t}"_corpus_tar.tar
   TARGET_BASE="$(expr "$t" : '.*/\(.*\)_fuzz_test')"
+  # There may be *.dict files in this folder that need to be moved into the OUT dir.
+  find "${CORPUS_UNTAR_PATH}" -type f -name *.dict -exec mv -n {} "${OUT}"/ \;
```
Comment (htuch) [inline projects/envoy/build.sh]: Why is there a `\;` on the end? Also, probably want to do `"*.dict"` to avoid potential shell match on any files in CWD.

Comment (asraa) [reply projects/envoy/build.sh]: The exec command ends at a `;` (man page: "All following arguments to find are taken to be arguments to the command until an argument consisting of `;` is encountered.") The example in the man page show the semi-colon escaped, and it seems by trail I have to do that as well.
